compiler/cpp/src/interpreter/Interpreter.cpp

```diff
@@ -1,40 +1,44 @@
-// Copyright (c) Marc Auberer 2021. All rights reserved.
-
 //
-// Created by Marc on 14.05.2021.
+// Created by Marc on 24.05.2021.
 //
 
-#include "interpreter.h"
+#include "Interpreter.h"
 
-bool getExistenceOfJsonKey(const std::unique_ptr<KeyExprAST> &key, json data) {
-    for (const std::unique_ptr<IdentifierExprAST>& identifier : key->getIdentifiers()) {
-        std::string identifierName = identifier->getName();
-        int identifierIndex = identifier->getIndex();
+Interpreter::Interpreter(bool isSingleStatement, const std::string &fileInput, const std::string &dataInput,
+                         const std::string &inputCommentLineIdentifiers,
+                         const std::string &inputCommentBlockOpenIdentifiers,
+                         const std::string &inputCommentBlockCloseIdentifiers) {
+    this->isSingleStatement = isSingleStatement;
 
-        if (!data.contains(identifierName)) return false;
-        data = data[identifierName];
+    // Parse input string from JSON to object
+    jsonParser = JSONParser(dataInput);
 
-        if (identifierIndex >= 0) { // Identifier has an index attached to it
-            if (data.size() <= identifierIndex) return false;
-            data = data[identifierIndex];
-        }
-    }
-    return true;
+    // Initialize analyzer
+    analyzer = Analyzer(isSingleStatement, fileInput, jsonParser, inputCommentLineIdentifiers,
+                        inputCommentBlockOpenIdentifiers, inputCommentBlockCloseIdentifiers);
+    ast = analyzer.getAST();
+
+    // Execute semantic analysis
+    analyzer.executeAnalysis();
 }
 
-std::string getOutput(bool isSingleStatement, TopLevelExprAST* ast, const json& data) {
+std::string Interpreter::interpretInput() {
+    return getOutput();
+}
+
+std::string Interpreter::getOutput() {
     if (isSingleStatement) {
         if (ast->getType() != TopLevelExprAST::STMT_LST_EXPR)
             throw std::runtime_error("Input was no single statement list");
         auto* stmtLst = static_cast<StmtLstExprAST*>(ast);
-        return evaluateStmtList(stmtLst, data) ? "true" : "false";
+        return evaluateStmtList(stmtLst) ? "true" : "false";
```

Comment:
`getOutput` returns the output string, that the user will get after implementing the whole file. In other words the end result. In the case of a single expression, this would be `true` or `false`. As the user wants to have a string the output file, we need to use `"true"` and `"false"` here.